dbms/src/AggregateFunctions/AggregateFunctionGroupUniqArray.cpp

```diff
@@ -20,17 +20,24 @@ class AggregateFunctionGroupUniqArrayDateTime : public AggregateFunctionGroupUni
 };
 
 
-static IAggregateFunction * createWithExtraTypes(const IDataType & argument_type)
+static IAggregateFunction * createWithExtraTypes(const DataTypes & argument_types)
 {
+    const IDataType& argument_type = *argument_types[0];
+
          if (typeid_cast<const DataTypeDate *>(&argument_type)) return new AggregateFunctionGroupUniqArrayDate;
     else if (typeid_cast<const DataTypeDateTime *>(&argument_type)) return new AggregateFunctionGroupUniqArrayDateTime;
     else
     {
+        IAggregateFunction* fun;
+
         /// Check that we can use plain version of AggreagteFunctionGroupUniqArrayGeneric
         if (argument_type.isValueUnambiguouslyRepresentedInContiguousMemoryRegion())
-            return new AggreagteFunctionGroupUniqArrayGeneric<true>;
+            fun = new AggreagteFunctionGroupUniqArrayGeneric<true>;
         else
-            return new AggreagteFunctionGroupUniqArrayGeneric<false>;
+            fun = new AggreagteFunctionGroupUniqArrayGeneric<false>;
+
+        fun->setArguments(argument_types);
```

Comment:
If the method will throw an exception, we get memory leak.

Reply:
It's ok for you if I put std::unique_ptr to avoid this ?

Reply:
Yes, exactly.